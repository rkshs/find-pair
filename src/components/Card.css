.card {
    position: relative;
    width: 120px;
    height: 135px;
    border: 1px solid black;
}

.card-image {
    position: absolute;
    width: 100%;
    height: 100%;
    background-image: url('../../public/icons.png');
    background-repeat: no-repeat;
    background-size: 700px;
    transform: rotateY(90deg);
    transition: all ease-in .2s;
}

.flip .card-image {
    transform: rotateY(0);
    transition-delay: .2s;
}

.card-cover {
    /*display: none;*/
    position: absolute;
    width: 100%;
    height: 100%;
    background-image: url('../../public/cover.png');
    background-repeat: no-repeat;
    background-size: 110px 120px;
    background-position: 5px 7px;
    transition: all ease-in .2s;
    transition-delay: .2s;
}

.flip .card-cover {
    transform: rotateY(-90deg);
    transition-delay: 0s;
}

.card-image-1 {
    background-position: 0 0;
}

.card-image-2 {
    background-position: -117px 0;
}

.card-image-3 {
    background-position: -235px 0;
}

.card-image-4 {
    background-position: -352px 0;
}

.card-image-5 {
    background-position: -470px 0;
}

.card-image-6 {
    background-position: -589px 0;
}

.card-image-7 {
    background-position: 0 -135px;
}

.card-image-8 {
    background-position: -117px -135px;
}

.card-image-9 {
    background-position: -235px -135px;
}

.card-image-10 {
    background-position: -352px -135px;
}

.card-image-11 {
    background-position: -470px -135px;
}

.card-image-12 {
    background-position: -589px -135px;
}

.card-image-13 {
    background-position: 0 -267px;
}

.card-image-14 {
    background-position: -117px -267px;
}

.card-image-15 {
    background-position: -235px -267px;
}

.card-image-16 {
    background-position: -352px -267px;
}

.card-image-17 {
    background-position: -470px -267px;
}

.card-image-18 {
    background-position: -589px -267px;
}