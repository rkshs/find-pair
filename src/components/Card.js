import './Card.css';

export default function Card({ card, setChoose, flip, disabled }) {

    const clickHandler = () => {
        if (!disabled) {
            setChoose(card)
        }
    }

    return (
        <div key={card.id} className={`card ${flip ? 'flip' : ''} ${card.matched ? 'matched' : ''}`}>
            <div className='card-cover' onClick={clickHandler} />
            <div className={`card-image ${card.className}`} />
        </div>
    )
}