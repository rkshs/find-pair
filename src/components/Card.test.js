import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import Card from './Card';

const cardMock = {
    className: 'card-image-11',
    matched: false,
    id: 0.6999073561281655
}

afterEach(cleanup);

describe('Card component', () => {
    it('snapshot renders correctly, truthy values', () => {
        const cardView = renderer
            .create(<Card
                key={cardMock.id}
                card={cardMock}
                setChoose={null}
                flip={false}
                disabled={false}
            />).toJSON()
        expect(cardView).toMatchSnapshot()
    });
    it('snapshot renders correctly, falsy values', () => {
        const cardView = renderer
            .create(<Card
                card={{className: '', matched: '', id: ''}}
            />).toJSON()
        expect(cardView).toMatchSnapshot()
    });
});

