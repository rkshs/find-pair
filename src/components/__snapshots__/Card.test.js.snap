// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component snapshot renders correctly, falsy values 1`] = `
<div
  className="card  "
>
  <div
    className="card-cover"
    onClick={[Function]}
  />
  <div
    className="card-image "
  />
</div>
`;

exports[`Card component snapshot renders correctly, truthy values 1`] = `
<div
  className="card  "
>
  <div
    className="card-cover"
    onClick={[Function]}
  />
  <div
    className="card-image card-image-11"
  />
</div>
`;
